### PR TITLE
Fix cache test for Spark 3.3.2 [databricks]

### LINKED ISSUE #7725: [BUG] cache_test failed w/ ParquetCachedBatchSerializer in spark 3.3.2-SNAPSHOT
**Describe the bug**
first saw at internal CI rapids_it-3.3.x-SNAPSHOT-dev-github build ID: 190

failed,
```

[2023-02-10T02:05:28.046Z] E                   Caused by: java.lang.IllegalArgumentException: For input string: "null"

[2023-02-10T02:05:28.046Z] E                   	at scala.collection.immutable.StringLike.parseBoolean(StringLike.scala:330)

[2023-02-10T02:05:28.046Z] E                   	at scala.collection.immutable.StringLike.toBoolean(StringLike.scala:289)

[2023-02-10T02:05:28.046Z] E                   	at scala.collection.immutable.StringLike.toBoolean$(StringLike.scala:289)

[2023-02-10T02:05:28.046Z] E                   	at scala.collection.immutable.StringOps.toBoolean(StringOps.scala:33)

[2023-02-10T02:05:28.046Z] E                   	at org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.<init>(ParquetSchemaConverter.scala:70)

[2023-02-10T02:05:28.046Z] E                   	at com.nvidia.spark.rapids.ParquetCachedBatchSerializer$CachedBatchIteratorConsumer$$anon$3.$anonfun$convertCachedBatchToInternalRowIter$1(ParquetCachedBatchSerializer.scala:744)

[2023-02-10T02:05:28.046Z] E                   	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)

[2023-02-10T02:05:28.046Z] E                   	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)

[2023-02-10T02:05:28.046Z] E                   	at com.nvidia.spark.rapids.ParquetCachedBatchSerializer.withResource(ParquetCachedBatchSerializer.scala:260)

[2023-02-10T02:05:28.046Z] E                   	at com.nvidia.spark.rapids.ParquetCachedBatchSerializer$CachedBatchIteratorConsumer$$anon$3.convertCachedBatchToInternalRowIter(ParquetCachedBatchSerializer.scala:729)

[2023-02-10T02:05:28.046Z] E                   	at com.nvidia.spark.rapids.ParquetCachedBatchSerializer$CachedBatchIteratorConsumer$$anon$3.hasNext(ParquetCachedBatchSerializer.scala:709)

[2023-02-10T02:05:28.046Z] E                   	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.hashAgg_doAggregateWithoutKey_0$(Unknown Source)

[2023-02-10T02:05:28.046Z] E                   	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)

[2023-02-10T02:05:28.046Z] E                   	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)

[2023-02-10T02:05:28.046Z] E                   	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)

[2023-02-10T02:05:28.047Z] E                   	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)

[2023-02-10T02:05:28.047Z] E                   	at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:140)

[2023-02-10T02:05:28.047Z] E                   	at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)

[2023-02-10T02:05:28.047Z] E                   	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)

[2023-02-10T02:05:28.047Z] E                   	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)

[2023-02-10T02:05:28.047Z] E                   	at org.apache.spark.scheduler.Task.run(Task.scala:136)

[2023-02-10T02:05:28.047Z] E                   	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:548)

[2023-02-10T02:05:28.047Z] E                   	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1504)

[2023-02-10T02:05:28.047Z] E                   	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:551)

[2023-02-10T02:05:28.047Z] E                   	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)

[2023-02-10T02:05:28.047Z] E                   	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
```

```
[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_cache_expand_exec[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'false'}-Float][IGNORE_ORDER, INCOMPAT]

[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_cache_partial_load[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'false'}-Array(String)][ALLOW_NON_GPU(CollectLimitExec)]

[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_cache_expand_exec[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'false'}-Double][IGNORE_ORDER, INCOMPAT]

[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_cached_join_filter[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'false'}-Boolean][IGNORE_ORDER, ALLOW_NON_GPU(ANY)]

[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_passing_gpuExpr_as_Expr[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'true'}][ALLOW_NON_GPU(CollectLimitExec)]

[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_cache_join[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'false'}-Timestamp][IGNORE_ORDER]

[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_cache_partial_load[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'true'}-Array(Decimal(20,2))][ALLOW_NON_GPU(CollectLimitExec)]

[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_cache_expand_exec[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'false'}-Boolean][IGNORE_ORDER]

[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_cache_partial_load[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'false'}-Array(Boolean)][ALLOW_NON_GPU(CollectLimitExec)]

[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_cached_join_filter[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'false'}-Date][IGNORE_ORDER, ALLOW_NON_GPU(ANY)]

[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_cache_expand_exec[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'false'}-Date][IGNORE_ORDER]

[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_passing_gpuExpr_as_Expr[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'false'}][ALLOW_NON_GPU(CollectLimitExec)]

[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_cache_partial_load[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'true'}-String][ALLOW_NON_GPU(CollectLimitExec)]

...

=========== 476 failed, 104 passed, 48 warnings in 176.98s (0:02:56) ===========
```

**Steps/Code to reproduce bug**
```
  PYSP_TEST_spark_sql_cache_serializer=com.nvidia.spark.ParquetCachedBatchSerializer \
    ./run_pyspark_from_build.sh -k cache_test
```

### PR DESCRIPTION
**Pull request by razajafri**
There has been a change in Spark 3.3.2 in SparkSchemaConverter where it now uses the `LEGACY_PARQUET_NANOS_AS_LONG` to determine whether to use TimestampType(nano) or to use long which loses the Timezone awareness. For us this is fine as we don't support TZ anyways. 

As discussed in [SPARK-40819](https://github.com/apache/spark/pull/39905) Timestamp(nano, true) will result in an `AnalysisException`. I don't think we need to highlight this in our documentation but I would like to know what others think. 

fixes #7725

### CODE REVIEW

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/ParquetCachedBatchSerializer.scala [hunk 1]
```diff
@@ -28,7 +28,7 @@ import ai.rapids.cudf.ParquetWriterOptions.StatisticsFrequency
 import com.nvidia.spark.GpuCachedBatchSerializer
```
Comment (tgravescs) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/ParquetCachedBatchSerializer.scala:28]: copyright needs to be updated with 2023

File: sql-plugin/src/main/332+/scala/com/nvidia/spark/rapids/shims/ParquetLegacyNanoAsLongShims.scala [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
```
Comment (tgravescs) [inline sql-plugin/src/main/332+/scala/com/nvidia/spark/rapids/shims/ParquetLegacyNanoAsLongShims.scala:2]: 2023

File: sql-plugin/src/main/311until332-all/scala/com/nvidia/spark/rapids/shims/ParquetLegacyNanoAsLongShims.scala [hunk 3]
```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
```
Comment (tgravescs) [inline sql-plugin/src/main/311until332-all/scala/com/nvidia/spark/rapids/shims/ParquetLegacyNanoAsLongShims.scala:2]: 2023

File: sql-plugin/src/main/332+/scala/com/nvidia/spark/rapids/shims/ParquetLegacyNanoAsLongShims.scala [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+package com.nvidia.spark.rapids.shims
+
+import org.apache.hadoop.conf.Configuration
+
+import org.apache.spark.sql.internal.SQLConf
+
+object ParquetLegacyNanoAsLongShims {
+  def setupLegacyParquetNanosAsLong(conf: Configuration): Unit = {
+    conf.setBoolean(SQLConf.LEGACY_PARQUET_NANOS_AS_LONG.key, true)
```
Comment (tgravescs) [inline sql-plugin/src/main/332+/scala/com/nvidia/spark/rapids/shims/ParquetLegacyNanoAsLongShims.scala:25]: why is the default conf setting true when spark has it false?

Comment (razajafri) [reply sql-plugin/src/main/332+/scala/com/nvidia/spark/rapids/shims/ParquetLegacyNanoAsLongShims.scala:25]: My thought was that we should just use long to save nano timestamps as we don't care about the timezone.

**Comment by razajafri** [comment]
build

**Review by tgravescs — APPROVED** [review]

**Pull request merged**